AirCasting/Settings/BackendURLValidator.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  BackendURLValidator.swift
+//  URLValidation
+//
+//  Created by Pawel Gil on 25/06/2021.
+//
+
+import Foundation
+
+class BackendURLBuilder {
+    enum ValidationError: Error {
+        case invalidURL
+        case invalidPort
+        case noURL
+    }
+    
+    func createURL(url: String, port: String) throws -> URL? {
+        if url.count == 0 && port.count == 0 { return nil }
+        if url.count == 0 && port.count > 0 { throw ValidationError.noURL }
+        let url = appenURLWithSchemeIfNotPresent(url: url)
+        var components = try getComponents(url: url)
+        try validateHost(for: components)
+        components.port = try getPortNumber(from: port)
+        return components.url!
+    }
+    
+    private func getComponents(url: String) throws -> URLComponents {
+        guard let components = URLComponents(string: url) else {
+            throw ValidationError.invalidURL
+        }
+        return components
+    }
+    
+    private func validateHost(for components: URLComponents) throws {
+        let host = try getHost(from: components)
+        if host.contains(".") {
+            let moreThanDot = components.host?.components(separatedBy: ".")[1]
+            guard host.contains(".") && moreThanDot!.count > 0 else {
+                throw ValidationError.invalidURL
+            }
+        } else {
+            guard host.contains(".") else {
+                throw ValidationError.invalidURL
+            }
+        }
+    }
+    
+    private func getHost(from components: URLComponents) throws -> String {
+        guard let host = components.host else {
+            throw ValidationError.invalidURL
+        }
+        return host
+    }
+    
+    private func getPortNumber(from port: String) throws -> Int? {
+        if port.count == 0 { return nil }
+        guard let portNumber = Int(port) else {
+            throw ValidationError.invalidPort
+        }
+        return portNumber
+    }
+    
+    private func appenURLWithSchemeIfNotPresent(url: String) -> String {
+        var newUrl = url
+        if !url.hasPrefix("http") {
```

Comment:
there is a pitfall here. Any url with prefix `http` will pass. Even this one: `httpSOMETHING.com`. Which is not ideal. I think we should change this `if` statement to this:

```Swift
if !(url.hasPrefix("http://") || url.hasPrefix("https://")) {
...
}
```

This solves the problem (which I believe is a problem indeed, I can imagine hosts which start with `http` like `httpfordummies.com` or whatever...